lab-teddy/server.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const Client = require('./model/client');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = net.createServer();
+const PORT = process.env.PORT || 8080;
+
+const pool = [];
+
+ee.on('default', (client, string) => {client.socket.write(`Not a valid command:${string.split(' ', 1)}`);
+});
+
+ee.on('@all', (client, string) => {
+  console.log('this is the string', string);
+  pool.forEach(c => c.socket.write(`${client.userName}: ${string}`));
+});
+
+ee.on('/nic', (client, string) => {
```

Comment:
If you're wondering why you can't set a nickname, it's because you are registering /nic here, but not referencing "/nic" in your server.on("data")...
      if(command.startsWith("/")

it doesn't know the difference between "/" and "@". I'd start there!